cmd/xfn/spark/spark.go

```diff
@@ -74,6 +75,8 @@ type Command struct {
 	CacheDir      string `short:"c" help:"Directory used for caching function images and containers." default:"/xfn"`
 	Runtime       string `help:"OCI runtime binary to invoke." default:"crun"`
 	MaxStdioBytes int64  `help:"Maximum size of stdout and stderr for functions." default:"0"`
+	DryRun        bool   `help:"pull images, but skip running fn container"`
```

Comment:
AFAIU, this flag is introduced for use only during e2e tests, right?
Could we instead have a test image doing nothing (e.g. available at [crossplane-e2e](https://github.com/crossplane/crossplane/issues/4257)), or always succeeding and we invoke it during tests?

Reply:
> AFAIU, this flag is introduced for use only during e2e tests, right?

I introduced so that we can have simpler integration tests, not e2e test.

> Could we instead have a test image doing nothing (e.g. available at https://github.com/crossplane/crossplane/issues/4257), or always succeeding and we invoke it during tests?

You mean a test fn image that does nothing? Sure, we could have them if we wanna have real e2e tests. Where we would manage and build such image? From this repo?

Reply:
I think it would be in a similar category as [provider-nop](https://github.com/crossplane-contrib/provider-nop), so we can have it in a separate repo.

Reply:
I agree with @turkenh here.

Reply:
As to where such an image would live - I think we can find a place. Today there's a few Configuration images used to test that I want to find a home for https://github.com/crossplane/crossplane/issues/4257. I think we could do the same for functions.